pkg/providerconfig/types.go

```diff
@@ -297,10 +297,7 @@ func (configVarResolver *ConfigVarResolver) GetConfigVarStringValueOrEnv(configV
 		return cfgVar, err
 	}
 
-	envVal, envValFound := os.LookupEnv(envVarName)
-	if !envValFound {
```

Comment:
Why did you remove this?

Reply:
Not property which gets loaded via this func is mandatory.
If we throw an error here, it is essentially mandatory.

Instead on relying on this implicit handling, we must put the validation of those fields into the corresponding "Validate" function.